# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Laraib175/FlyRank-week1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


I pick Lane 2: Refresh / Content Opportunity Scoring.
I chose this lane because it directly addresses a real operational need—helping SEO teams decide which pages to review and rewrite first.
The starter pipeline already showed me that a learned model (random forest) can achieve a 0.740 Precision@50, which is a ~3x lift over the baseline hand-rule (0.240),
so I believe there is genuine signal to uncover and improve upon for a more robust ranking system.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [6]:
# This cell is for CODE (numbers, a query, a check).

# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


Decision: Which pages should be sent to the content review team for refresh, expansion, or protection?

Action: The content team will take the top-ranked pages from my queue and perform specific actions (rewrite, update statistics, improve internal linking, or monitor) based on the reason codes I provide.

Cost of a wrong call:

False Positive (wasting time): A writer spends hours rewriting a page that didn't actually need it, wasting limited capacity and delaying work on genuinely declining pages.

False Negative (missing a problem): A page that is quietly losing traffic gets ignored, and by the time someone notices, it has dropped off the first page of search results and lost significant revenue potential.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
# Step 1: Clone your repo into Colab's temporary machine
# 🔴 REPLACE 'your-username/your-repo-name' with your actual GitHub username and repo name!
# Example: if your repo URL is https://github.com/Laraib-01/my-internship, write 'Laraib-01/my-internship'
!git clone https://github.com/Laraib175/FlyRank-week1.git repo
%cd repo

# Step 2: Now load the dataset
import pandas as pd

df = pd.read_csv('data/raw/content_refresh_anonymized.csv')

# Step 3: Show 3 real numbers that support Lane 2
print("1. Total pages in the starter dataset:", len(df))
print("2. Pages currently flagged as declining (trend_direction == 'down'):",
      (df['trend_direction'] == 'down').sum())
print("3. Percentage of pages with at least 500 impressions (high visibility):",
      round((df['impressions_90d'] >= 500).mean() * 100, 1), "%")

Cloning into 'repo'...
remote: Enumerating objects: 122, done.
remote: Counting objects: 100% (122/122), done.
remote: Compressing objects: 100% (78/78), done.
remote: Total 122 (delta 35), reused 98 (delta 28), pack-reused 0 (from 0)
Receiving objects: 100% (122/122), 1.86 MiB | 9.93 MiB/s, done.
Resolving deltas: 100% (35/35), done.
/content/repo/repo/repo
1. Total pages in the starter dataset: 30000
2. Pages currently flagged as declining (trend_direction == 'down'): 16262
3. Percentage of pages with at least 500 impressions (high visibility): 55.8 %


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [8]:
# This cell is for CODE (numbers, a query, a check).

# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


What I CAN claim:

I can build a ranking system that prioritizes pages showing early signs of decline or opportunity, based on observed search and engagement signals.

I can measure whether my model's ranking beats a simple hand-rule baseline using Precision@K metrics on a client-holdout validation set.

What I CANNOT claim:

I cannot claim that editing or refreshing these pages will cause them to recover. My data is observational—I see correlations, not causation.

I cannot claim this model works for the full 79M-row warehouse unless I re-validate it against that full dataset with a proper time-aware split.

I cannot claim SEO algorithm factors, Google ranking signals, or AI citations—the dataset only contains observable measurements of past performance.

☑ ✅ Have I named my lane? (Yes)
☑ ✅ Have I named the decision this work supports? (Yes)
☑ ✅ Have I named the action someone takes? (Yes)
☑ ✅ Have I named the cost of a wrong recommendation? (Yes)
☑ ✅ Have I shown at least 2 real numbers from the starter data? (Yes—30,000 total, 16,262 declining, 55.8% high-visibility)
☑ ✅ Have I explained why this is not just "train a model"? (Yes—it's about ranking and decision support)